tcms/testruns/tests/test_permissions.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=invalid-name, too-many-ancestors
+
+from http import HTTPStatus
+
+from django.urls import reverse
+
+from tcms.tests import PermissionsTestCase
+from tcms.tests import remove_perm_from_user
+from tcms.tests import user_should_have_perm
+from tcms.tests import factories
+
+
+class TestEditRun(PermissionsTestCase):
+    """Test edit view method"""
+
+    permission_label = 'testruns.change_testrun'
+    http_method_names = ['get', 'post']
+    url = 'url'
+    post_data = {
+        'summary': '',
+        'build': '',
+        'manager': '',
+        'default_tester': '',
+        'notes': '',
+    }
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        cls.product = factories.ProductFactory(name='KiwiTCMS')
+        cls.build = factories.BuildFactory(product=cls.product)
+        cls.version = factories.VersionFactory(value='0.1', product=cls.product)
+        cls.plan = factories.TestPlanFactory(
+            author=cls.tester,
+            product=cls.product,
+            product_version=cls.version)
+
+        cls.test_run = factories.TestRunFactory(product_version=cls.version,
+                                                plan=cls.plan,
+                                                build=cls.build,
+                                                manager=cls.tester,
+                                                default_tester=cls.tester)
+
+        cls.url = reverse('testruns-edit', args=[cls.test_run.pk])
+        cls.intern = factories.UserFactory(username='intern',
+                                           email='intern@example.com')
+
+    def verify_get_with_permission(self):
+        user_should_have_perm(self.tester, 'testruns.change_testrun')
+        response = self.client.get(self.url)
+
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertContains(
+            response,
+            '<input type="text" id="id_summary" name="summary" value="%s" class="form-control"'
+            ' required>' % self.test_run.summary,
+            html=True
+        )
+
+    def verify_post_with_permission(self):
+        user_should_have_perm(self.tester, 'testruns.change_testrun')
+
+        post_data = {
+            'summary': 'New run summary',
+            'build': self.build.pk,
+            'manager': self.test_run.manager.email,
+            'default_tester': self.intern.email,
+            'notes': 'easytest',
+        }
+
+        self.assertRedirects(
+            self.client.post(self.url, post_data),
+            reverse('testruns-get', args=[self.test_run.pk]))
```

Comment:
Break this down into `response = self.client.post() && asserts`. 

Will also have to refresh the object from the DB and make sure the new values match what is edited and doesn't match what was there before to make sure editing actually worked.